### DISCUSSION TITLE
getaddrinfo is taking longer

### DISCUSSION THREAD

**Discussion by paresy** [baresip/baresip#2698] [General]
Ref: https://github.com/baresip/baresip/discussions/2696#discussioncomment-6877349

Just for clarification: getaddrinfo is successful - it is just taking longer. (~5 seconds)

My test code:

```
#include <sys/socket.h>
#include <sys/time.h>
#include <netdb.h>
#include <cstdio>
#include <cstring>

void printTS(const char* name) {
    struct timeval te;
    gettimeofday(&te, NULL);
    long long milliseconds = te.tv_sec*1000LL + te.tv_usec/1000;
    printf("%s: %lld\n", name, milliseconds);
}

int main() {
    struct addrinfo *res0 = NULL;
    struct addrinfo hints;
    int err;

    printTS("start");
    err = getaddrinfo("invalid.local", NULL, NULL, &res0);
    printTS("end  ");
    printf("------------ ERR: %d\n", err);

    printTS("start");
    err = getaddrinfo("6c61b18a-7b57-45aa-8208-39e2303198f8.local", NULL, NULL, &res0);
    printTS("end  ");
    printf("------------ ERR: %d\n", err);

    memset(&hints, 0, sizeof(hints));
    hints.ai_family = AF_INET;
    printTS("start");
    err = getaddrinfo("6c61b18a-7b57-45aa-8208-39e2303198f8.local", NULL, &hints, &res0);
    printTS("end  ");
    printf("------------ ERR: %d\n", err);

    return 0;
}

```

Result:

```
start: 1693513291283
end  : 1693513296287
------------ ERR: 8
start: 1693513296287
end  : 1693513301297
------------ ERR: 0
start: 1693513301297
end  : 1693513301297
------------ ERR: 0
```

-> MacOS Ventura, IPv6 is enabled, but my router is not serving proper IPv6 addresses through DHCPv6.

Output of ifconfig for the relevant interface:

```
en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether f8:4d:89:5e:b0:07 
	inet6 fe80::8f7:8374:a89b:7e5c%en0 prefixlen 64 secured scopeid 0xe 
	inet 192.168.1.89 netmask 0xffffff00 broadcast 192.168.1.255
	inet6 fd4b:a493:d8a7:4a5e:1823:55e6:2bdc:2ae7 prefixlen 64 autoconf secured 
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active

```

The .local address was exposed by Firefox on the same machine. When i do the same with Safari on my iPhone and try to use getaddrinfo on this .local address it returns immediately.

**Comment by sreimers** [comment]
Ok looks like this a known mDNS macOS issue: 
https://github.com/rust-lang/rust/issues/31665
https://stackoverflow.com/questions/44760633/mac-os-x-slow-connections-mdns-4-5-seconds-bonjour-slow

I will try to optimize our code to avoid the startup delay (only necessary with no other candidates). And we can resolve IPv4 and IPv6 with two separate `getaddrinfo` calls.

**Comment by sreimers** [comment]
Should be fixed by: https://github.com/baresip/re/commit/ad4de14e13145561529170c4f746fc3cb62b2829
